src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilterTest.java

```diff
@@ -310,12 +311,45 @@ public void testEqualsAndHashCodeOfTagClass() {
 
     @Test
     public void testToStringOfTagClass() {
-        final SuppressionCommentFilter filter = new SuppressionCommentFilter();
-        final Object tag =
-                getTagsAfterExecution(filter, "filename", "//CHECKSTYLE:OFF").get(0);
+        final SuppressionCommentFilter filter1 = new SuppressionCommentFilter();
+        final Object tag1 =
+                getTagsAfterExecution(filter1, "filename", "//CHECKSTYLE:OFF").get(0);
         assertEquals("Invalid toString result",
             "Tag[text='CHECKSTYLE:OFF', line=1, column=0, type=OFF,"
-                    + " tagCheckRegexp=.*, tagMessageRegexp=null]", tag.toString());
+                    + " tagCheckRegexp=.*, tagMessageRegexp=null]", tag1.toString());
+
+        final SuppressionCommentFilter filter2 = new SuppressionCommentFilter();
+        filter2.setMessageFormat(".*");
+        final Object tag2 =
+                getTagsAfterExecution(filter2, "filename", "//CHECKSTYLE:ON").get(0);
+        assertEquals("Invalid toString result",
+            "Tag[text='CHECKSTYLE:ON', line=1, column=0, type=ON,"
+                + " tagCheckRegexp=.*, tagMessageRegexp=.*]", tag2.toString());
+    }
+
+    @Test
+    public void testCompareToOfTagClass() {
```

Comment:
for complete coverage of `compareTo`, we need to compare two tags on the same line